packages/notifier/README.md

```diff
@@ -1,6 +1,263 @@
+remains correct, but Alice and Bob may each have missed either or both of the
+non-final values.
+
+
+## Distributed Operation
+
+Both pairs can be used in a
+multicast manner with good distributed systems properties, where there is only
+one producing site but any number of consuming sites. The producer is not
+vulnerable to the consumers and the consumers are not vulnerable to each other.
+For distributed operation, all the iteration values---non-final values,
+successful completion value, failure reason---must be `Passable`, i.e., must be
+values that can somehow be passed betwen vats. The rest of this page assumes all
+these values are Passable.
+
+The maker is used to make the pair on the producer's site. As a result, both
+the iterationObserver and the initial asyncIterable are on the producer's site.
+If Paula the producer sends Bob the asyncIterable, Bob receives a possibly
+remote reference to the async iterable Bob's code above is still correct if he
+uses this
+reference directly, since `observeIteration` only needs its first argument to
+be an `ERef<AsyncIterable<Passable>>`, i.e., a reference of some sort to an
+AsyncIterable conveying Passable values. This reference may be a local
+AsyncIterable, a remote presence of an AsyncIterable, or a local or remote
+promise for an AsyncIterable. `observeIteration` only sends it eventual
+messages using `E` (equivalent to the tildot syntax `~.`), and so doesn't care
+about these differences.
+
+Although Bob's code is correct, it is sub-optimal. Its distributed systems
+properties are not terrible, but Bob can do better using the
+`getSharableSubsciptionInternals()` method provided by both NotifierKit and
+SubscriptionKit. This enables consumer Bob to make a local AsyncIterable that
+coordinates better with producer Paula's IterationObserver. Alice's code is
+less forgiving. She's using JavaScript's `for-await-of` loop which requires a
+local AsyncIterable. It cannot handle a remote reference to an AsyncIterable
+at
+Paula's site. Alice has no choice but to make an AsyncIterable at her site.
+Using `getSharableSubsciptionInternals()` is the best way for her to do so. She
+can replace her last line, the call to `consume(subscription)` with
+
+```js
+import { makeSubscription } from '@agoric/notifier';
+
+const localSubscription =
+  makeSubscription(E(subscription).getSharableSubsciptionInternals());
+consume(localSubscription);
+```
+
+For NotifierKits there is a similar pair of a `getSharableNotifierInternals`
+method and a `makeNotifier`. However, the technique requires Alice to know what
+kind of possibly-remote AsyncIterable she has, and to have the required making
+function code locally available. Alternatively, Alice can generically mirror
+any possibly remote AsyncIterable by making a new local pair and plugging them
+together with `observeIteration`.
+```js
+const { publication: p, subscription: s } = makeSubscriptionKit();
+observeIteration(subscription, p);
+consume(s);
```

Comment:
I think this example is easier to read with full names, without abbreviations.

Reply:
Needed non-colliding names so went with `adapterPublication` and `adapterSubscription`. Done